stac_fastapi/types/stac_fastapi/types/core.py

```diff
@@ -581,9 +581,7 @@ async def landing_page(self, **kwargs) -> stac_types.LandingPage:
                 "rel": "service-desc",
                 "type": "application/vnd.oai.openapi+json;version=3.0",
                 "title": "OpenAPI service description",
-                "href": urljoin(
-                    str(request.base_url), request.app.openapi_url.lstrip("/")
-                ),
+                "href": urljoin(base_url, request.app.openapi_url.lstrip("/")),
```

Comment:
When a user adds a router_prefix to the API, the `base_url` as the prefix added in https://github.com/stac-utils/stac-fastapi/blob/f54227871c29cf15ae49b2ab96b28f357de60485/stac_fastapi/types/stac_fastapi/types/requests.py#L6-L14

The issue is that, when we create the fastapi app 

https://github.com/stac-utils/stac-fastapi/blob/f54227871c29cf15ae49b2ab96b28f357de60485/stac_fastapi/api/stac_fastapi/api/app.py#L75-L85

we set the openapi_url to either `/api` or to what the user has set the `openapi_url` settings.

so by default we register the openapi to `/api` but here when we do `urljoin(base_url, request.app.openapi_url.lstrip("/")),`, we endup with `/{base_url}/{router_prefix}/api`